src/Avalonia.Controls/Button.cs

```diff
@@ -207,6 +208,11 @@ protected override void OnDetachedFromVisualTree(VisualTreeAttachmentEventArgs e
 
         protected override void OnAttachedToLogicalTree(LogicalTreeAttachmentEventArgs e)
         {
+            if (_hotkey != null) // Control attached again, set Hotkey to create a hotkey manager for this control
+            {
+                HotKey = _hotkey;
```

Comment:
Not a huge fan of this because `HotKey` may have been set with a non-LocalValue priority and here it's being promoted to `LocalValue` priority. This probably won't be noticed most of the time because non-LocalValue values are usually set by styles which will be detached from the control when it leaves the logical tree, but I suspect it's likely to cause hard-to-diagnose problems down the line.

Reply:
@grokys Thanks for replying, if this changes anything - I have been using this code in my production app for about 3 months, didn't notice any problem with styles not applying (I guess this might happen when changing themes and my app supports that).

I understand that because both the HotkeyManager and the Control itself are managing the Hotkey it isn't as clean, but as far as I understand other solution would be to refactor the HotkeyManager and I understood you wouldn't want that.

Also, without this putting control that uses HotKey inside templates (like button inside ListBox) will cause **drastic** memory leak, and obviously I can't ship my app like that.
If there are any solutions you guys would accept, then I will gladly implement it.